### DISCUSSION TITLE
Minitest support

### DISCUSSION THREAD

**Discussion by TastyPi** [ElMassimo/capybara-compose#25] [Ideas]
I've been looking for something to make the page object pattern easier, and this looks like exactly what I want! Unfortunately it seems to only support RSpec for now.

Would you be open to receiving pull requests for adding Minitest support? I haven't dug deep into the code to see how much work that would be, wanted to check that would be something you'd be open to before I spend too much time investigating.

**Comment by ElMassimo** [comment]
[abstraction]: https://github.com/ElMassimo/capybara-compose/blob/main/lib/capybara_test_helpers/config.rb#L51

Hi Graham!

As long as it doesn't require changes to the public API, and to the [definition](https://github.com/ElMassimo/capybara-compose/blob/main/lib/capybara_test_helpers/matchers.rb) of methods, I think it would be interesting to support Minitest.

Mostly, I don't want to increase the maintenance burden by tracking the list of methods twice, as I don't use Minitest myself and don't have a large (real) codebase where I can test upgrades.

---

Something that might make it possible to support Minitest with relatively minimal changes, is that for the most part there's a [single abstraction][abstraction] for how the methods are delegated to Capybara.

Assuming that Capybara follows the same naming pattern for RSpec and Minitest helpers, it would be possible to change the [generated code][abstraction] from:

```ruby
"#{ assertion ? "expect(#{ target }).to_or not_to, test_context" : target }.#{ method_name }(*args, **kwargs, &filter)"
```

to something like:

```ruby
"#{ target }.send('#{ assertion ? not_to ? 'wont_' : 'must_' : '' }#{ method_name }', *args, **kwargs, &filter)"
```

Then, it would be a matter of replacing `require 'capybara/rspec'` with a helper that auto-detects whether it's Minitest or RSpec.

If this works as intended, the advantage is that the public-facing API stays unchanged, so the [documentation](https://capybara-compose.netlify.app/) is valid for both.

**Reply by TastyPi** [reply]
Yeah my rough plan if I get round to it was to make just the minimal changes needed to add minitest support while not changing the API (if possible). Not sure when/if I'll have time to look into it, but glad to know you're willing to accepts PRs for it.
